### PR TITLE
[IMP] ADD COLUMN IF NOT EXISTS department_id

### PR DESCRIPTION
**Pull request by mileo**
Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why it exists?

**Comment by mileo** [comment]
Because of this old module:

https://github.com/OCA/department/blob/4a8d54b4a1a8ea58eb65ef1ea300eb40477e8511/analytic_base_department/analytic.py#L38

**Comment by pedrobaeza** [comment]
OK, thanks for clarifying. Then maybe next sql instruction shouldn't be executed in that case, isn't it?

**Comment by mileo** [comment]
Well noted, i will fix it tomorrow. 

Thanks

**Comment by pedrobaeza** [comment]
Any news on this?

**Comment by mileo** [comment]
@gabrielcardoso21 will continue this week.

File: addons/hr_timesheet/migrations/10.0.1.0/pre-migration.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 def create_and_populate_department(cr):
     cr.execute('''
-       ALTER TABLE account_analytic_line ADD COLUMN department_id INT;
+       ALTER TABLE account_analytic_line ADD COLUMN IF NOT EXISTS department_id INT;
```
Comment (StefanRijnhart) [inline addons/hr_timesheet/migrations/10.0.1.0/pre-migration.py:20]: Thanks! Can you add a docstring to this method mentioning the OCA module that provided the field for 9.0?

**Comment by pedrobaeza** [comment]
Any news on this?

**Comment by gabrielcardoso21** [comment]
@pedrobaeza I made a change to only update the account_analytic_line.department_id if the field department_id is null. Doesn't it solve the problem?

**Comment by pedrobaeza** [comment]
@gabrielcardoso21 https://github.com/OCA/OpenUpgrade/pull/1809#discussion_r298890702 is pending about adding a comment in the code.

**Comment by gabrielcardoso21** [comment]
@pedrobaeza @StefanRijnhart I made a commit adding the docstring required.

**Review by StefanRijnhart — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
That's it! Thank you.

**Pull request merged**